### PR TITLE
Table loading skeleton

### PR DESCRIPTION
**Pull request by tanercura**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [x] Enhancement feature

### 💡 Background and solution

1. Updated the Table loading flag to render skeleton loading instead of a generic spinner. This (imo) has a modern, better look and feel.

2. Evidence: 
https://user-images.githubusercontent.com/30558171/193475911-0faffaba-afeb-4beb-8bb9-0962cd0e7883.mp4

3. Usage is simply `<Table loading={{ type: 'skeleton', loading: true }} {...props} />`

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |  Refactored table loading flag to display skeleton loader  |
| 🇨🇳 Chinese |           |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Comment by afc163** [comment]
Cool! It is better to make it optional.

```jsx
<Table loading={{ type: 'spin' | 'sketeton', ...props }} />
```

**Comment by TnRc** [comment]
@afc163 done

File: components/table/index.en-US.md [hunk 1]
```diff
@@ -160,6 +160,18 @@ One of the Table `columns` prop for describing the table's columns, Column has t
...
+
+Properties for loading.
+
+| Property | Description | Type | Default |
+| --- | --- | --- | --- |
+| type | Loading component to use. Options: `spin` and `skeleton` | string | `spin` |
+
+For more on props to configure `spin` please check [`Spin`](/components/spin/).
+
+For more on props to configure `skeleton` please check [`Skeleton`](/components/skeleton/).
```
Comment (afc163) [inline components/table/index.en-US.md:173]: components/table/index.zh-CN.md

File: components/table/demo/dynamic-settings.md [hunk 2]
```diff
@@ -280,6 +290,7 @@ const App: React.FC = () => {
       </Form>
       <Table
         {...tableProps}
+        loading={loadingType == 'skeleton' ? { type: 'skeleton', loading }: { type: 'spin', spinning: loading }}
```
Comment (afc163) [inline components/table/demo/dynamic-settings.md:293]:
```suggestion
        loading={loadingType == 'skeleton' ? { type: 'skeleton', loading } : { type: 'spin', spinning: loading }}
```

File: components/table/Table.tsx [hunk 3]
```diff
@@ -69,6 +69,16 @@ interface ChangeEventInfo<RecordType> {
...
 }
 
+export interface TableSkeletonProps extends SkeletonProps {
+  /** Count for number of table rows to show */
+  rows?: number;
+}
+
+export interface TableLoadingProps extends SpinProps, TableSkeletonProps {
+  /** Type of loading component to use - defaulting to spin */
+  type?: 'spin' | 'skeleton';
```
Comment (afc163) [inline components/table/Table.tsx:79]:
```suggestion
  component?: 'spin' | 'skeleton';
```

**Comment by afc163** [comment]
Please check ci

File: components/table/index.en-US.md (hunk 1 above)
Comment (TnRc) [reply components/table/index.en-US.md:173]: @afc163 is there someone that can help me with the translation or do I just google translate?

**Comment by MadCcc** [comment]
New feature should be base on `feature` branch。

**Comment by jAtlasCode** [comment]
I have built custom table components for inner tables to match column counts... excited about this one! Great work

**Comment by afc163** [comment]
ci failed

**Comment by afc163** [comment]
ci still failed

**Comment by afc163** [comment]
ping

**Comment by TnRc** [comment]
@afc163 Been busy - I will try taking a look this weekend

**Comment by afc163** [comment]
https://github.com/ant-design/ant-design/actions/runs/3170106793/jobs/5197692884

**Comment by afc163** [comment]
still broken

**Comment by afc163** [comment]
https://github.com/ant-design/ant-design/actions/runs/3267839489/jobs/5378364863

**Comment by afc163** [comment]
Please rebase master.

**Comment by afc163** [comment]
图片

**Comment by afc163** [comment]
Please send it to feature branch.

**Pull request closed without merging**